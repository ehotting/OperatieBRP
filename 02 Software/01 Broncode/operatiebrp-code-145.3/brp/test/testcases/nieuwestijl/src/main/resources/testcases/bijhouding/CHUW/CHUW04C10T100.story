Meta:
@status                 Klaar
@usecase                UCS-BY.HG

Narrative: Verwerking Correctie Huwelijk

Scenario:   Het laten herleven van een ontbonden huwelijk
            LT: CHUW04C10T100

Given alle personen zijn verwijderd
Given enkel initiele vulling uit bestand /LO3PL-CHUW/CHUW04C10T100-001.xls
Given enkel initiele vulling uit bestand /LO3PL-CHUW/CHUW04C10T100-002.xls

!-- Voltrekking van een huwelijk
When voer een bijhouding uit CHUW04C10T100a.xml namens partij 'Gemeente BRP 1'
Then heeft het antwoordbericht verwerking Geslaagd

Given pas laatste relatie van soort 1 aan tussen persoon 300039001 en persoon 541737673 met relatie id 30010001 en betrokkenheid id 30010001
And de database is aangepast met: update kern.his_relatie set id = 9999 where id = (select hr.id from kern.his_relatie hr join kern.relatie r on r.id = hr.relatie where r.srt = 1 and hr.dataanv = 20160510)

Then is in de database de persoon met bsn 300039001 wel als PARTNER betrokken bij een HUWELIJK
Then is in de database de persoon met bsn 541737673 wel als PARTNER betrokken bij een HUWELIJK

!-- Ontbinding huwelijk
When voer een bijhouding uit CHUW04C10T100b.xml namens partij 'Gemeente BRP 1'
Then heeft het antwoordbericht verwerking Geslaagd

Then is in de database de persoon met bsn 300039001 niet als PARTNER betrokken bij een HUWELIJK
Then is in de database de persoon met bsn 541737673 niet als PARTNER betrokken bij een HUWELIJK

!-- De niet-vervallen rij gereed maken om als voorkomensleutel te gebruiken
Given de database is aangepast met: update kern.his_relatie
                                    set    id=9998
				    where  relatie=30010001
				    and    id<>9999
				    and    tsverval is null

!-- Correctie registratie huwelijk
When voer een bijhouding uit CHUW04C10T100c.xml namens partij 'Gemeente BRP 1'

Then heeft het antwoordbericht verwerking Geslaagd

Then is in de database de persoon met bsn 300039001 wel als PARTNER betrokken bij een HUWELIJK
Then is in de database de persoon met bsn 541737673 wel als PARTNER betrokken bij een HUWELIJK

Then is het antwoordbericht gelijk aan /testcases/bijhouding/CHUW/expected/CHUW04C10T100.xml voor expressie //brp:bhg_hgpCorrigeerHuwelijkGeregistreerdPartnerschap_R
