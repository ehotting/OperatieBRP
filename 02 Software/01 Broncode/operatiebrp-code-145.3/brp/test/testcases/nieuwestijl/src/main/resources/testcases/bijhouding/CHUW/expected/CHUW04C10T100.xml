<brp:bhg_hgpCorrigeerHuwelijkGeregistreerdPartnerschap_R xmlns:brp="http://www.bzk.nl/brp/brp0200">
    <brp:stuurgegevens>
        <brp:zendendePartij>199901</brp:zendendePartij>
        <brp:zendendeSysteem>BRP</brp:zendendeSysteem>
        <brp:referentienummer>*</brp:referentienummer>
        <brp:crossReferentienummer>00000000-0000-0000-0042-200000000104</brp:crossReferentienummer>
        <brp:tijdstipVerzending>*</brp:tijdstipVerzending>
    </brp:stuurgegevens>
    <brp:resultaat>
        <brp:verwerking>Geslaagd</brp:verwerking>
        <brp:bijhouding>Verwerkt</brp:bijhouding>
        <brp:hoogsteMeldingsniveau>Geen</brp:hoogsteMeldingsniveau>
    </brp:resultaat>
    <brp:correctieHuwelijk brp:objecttype="AdministratieveHandeling">
        <brp:partijCode>507013</brp:partijCode>
        <brp:tijdstipRegistratie>*</brp:tijdstipRegistratie>
        <brp:bijgehoudenPersonen>
            <brp:persoon brp:objecttype="Persoon">
                <brp:identificatienummers>
                    <brp:burgerservicenummer>300039001</brp:burgerservicenummer>
                </brp:identificatienummers>
            </brp:persoon>
            <brp:persoon brp:objecttype="Persoon">
                <brp:identificatienummers>
                    <brp:burgerservicenummer>541737673</brp:burgerservicenummer>
                </brp:identificatienummers>
            </brp:persoon>
        </brp:bijgehoudenPersonen>
        <brp:bijhoudingsplan brp:objecttype="Bijhoudingsplan">*</brp:bijhoudingsplan>
    </brp:correctieHuwelijk>
</brp:bhg_hgpCorrigeerHuwelijkGeregistreerdPartnerschap_R>
